packages/atclient/tests/test_atserver_message.c

```diff
@@ -233,7 +238,7 @@ uint8_t test_5b_bad_parse_heap() {
   expect_uint("len", 0, message.len);
 
   // ensure original heap buffer is intact
-  if (strncmp(heap_buffer, static_buffer, len) != 0) {
+  if (strncmp(heap_buffer, static_buffer, len - 1) != 0) {
```

Comment:
-1 to avoid comparing the `\n` which now should be replaced with a null-terminator